File: T34-Tuplespaces-2025/Client-Python/client_debug.py
import sys
class ClientDebug:
    _debug_flag = False
    # Construtor privado para impedir a criação de instâncias
    def __new__(cls, *args, **kwargs):
        raise NotImplementedError("ClientDebug não pode ser instanciada.")

    @staticmethod
    def setDebugFlag(val: bool):
        """Define o estado da flag de debug."""
        ClientDebug._debug_flag = val

    @staticmethod #ver
    def debug(debugMessage: str):
        """Imprime mensagens de debug se a flag estiver ativada."""
        if ClientDebug._debug_flag:
            print(debugMessage, file=sys.stderr)

File: T34-Tuplespaces-2025/Client-Java/src/main/java/pt/ulisboa/tecnico/tuplespaces/client/ClientDebug.java
package pt.ulisboa.tecnico.tuplespaces.client;

public final class ClientDebug {

    private static boolean debug_flag;

    private ClientDebug() {        
    }
    
    public static void setDebugFlag(boolean val){
        debug_flag=val;
    }

    public static void debug(String debugMessage){
        if(debug_flag)
            System.err.println(debugMessage);
    }
}